### DISCUSSION TITLE
Cannot abort failed migration

### DISCUSSION THREAD

**Discussion by riolly** [geldata/gel#4054] [Q&A] [answered]
After encountering an error on migration (using [migration ddl](https://www.edgedb.com/docs/reference/ddl/migrations)), I cannot cancel the migration using `abort migration` command.
```
error: EdgeQLSyntaxError: Unexpected 'migration'

dracula[tx:failed]> abort migration;
edgedb error: TransactionError: current transaction is aborted, commands ignored until end of transaction block
dracula[tx:failed]>
```
So, what I do is terminate the CLI.
When the migration does not error, `abort migration` works just fine.
Any suggestion?

**Comment by tailhook** [comment, marked as answer]
You can do `rollback` to rollback the whole transaction. Any specific reason you want to do `abort migration` without rolling back transaction? (in this case, you can use savepoints)

Background: when some query fails in the transaction (you can notice that state in shell as `[tx:failed]`) we don't allow any query to execute (exceptions are `rollback` and `rollback to savepoint`). In this sense `abort migration` is no different from any other query.

**Reply by elprans** [reply]
I think we should treat `abort migration` as `rollback to savepoint`/`rollback`, since that's part of what's happening internally.

**Reply by riolly** [reply]
Wow, thank you. I need to learn more about the migration concept.
